### ISSUE TITLE
pgvector dim size 

### ISSUE THREAD

**Issue by nathan-vo810** [superlinked/VectorHub#513]
The limit for the vector type is 16,000 dimensions ([docs](https://github.com/pgvector/pgvector?tab=readme-ov-file#vector-type)). 2,000 is the limit for indexing it (you'll see an error if you try).

**Comment by dhruv-anand-aintech** [comment]
> 2,000 is the limit for indexing it

I think it would be reasonable to keep it as 2000 then. A viewer of this table should not have to try vector search with a 2k+ vector on pgvector and then see the error after we listed 16k.

At max, it can be added as a note in the comment section of the cell.

**Comment by nathan-vo810** [comment]
Here's a refined version of your message:

"As a user, when I first saw the vector size limit, I assumed it wouldn’t be possible to store embeddings larger than 2000 dimensions. However, it turns out it can, and the vector search (nearest neighbor search) works just fine. 

The only aspect affected by the 2000-dimension limit is indexing. 

In any case, it’s up to you to decide how to handle this. Thanks for providing such a comprehensive tool for comparison!"

**Comment by nathan-vo810** [comment]
In my current use case, I’m using Langchain with pgvector, specifically
with the text-3-embedding-large model, which has a vector dimension of
3072. I’m able to perform similarity searches on the database without any
issues.

The 2,000 limit only applies to indexing, so it doesn't prevent vector
searches. What I'm trying to clarify is that the vector dim column in the
table should have a dimension of 16,000 to match the vector type in
pgvector.

**Comment by svonava-superlinked** [comment]
Got it! How many vectors do you have and what search latency do you observe?

**Comment by nathan-vo810** [comment]
I don’t actually measure the latency, but with my current DB of 15,000
records, I think it’s less than 1s.

**Comment by svonava-superlinked** [comment]
@nathan-vo810 got it - and probably quite low query-per-second? as in, sub 1 QPS?

@dhruv-anand-aintech I think the proper way to handle this would be to have dim limit per indexing algorithm (since now we have a list of supported algos), but that sounds like a nightmare to maintain..

alternatively, we add a comment for the dims column to clarify that this is for the ANN-type indexes.

What do you think?

**Comment by dhruv-anand-aintech** [comment]
Yeah I would prefer the latter suggestion (clarify further in column description), as this kind of case is not common.
